packages/components/src/components/FiltersForm.vue

```diff
@@ -80,27 +103,90 @@ export default {
   },
 
   methods: {
+    getSuggestionClasses(index) {
+      const classes = ['filters__form-suggestions-item'];
+      if (index === this.selectedSuggestion) {
+        classes.push('filters__form-suggestions-item--selected');
+      }
+      return classes;
+    },
     onInputFocus() {
       this.showSuggestions = true;
     },
     onFormSubmit() {
+      if (
+        this.$refs.input === window.document.activeElement &&
+        this.showSuggestions &&
+        this.selectedSuggestion !== null
+      ) {
+        this.inputValue = this.suggestionsList[this.selectedSuggestion];
+      }
+
       const value = this.inputValue;
       this.showSuggestions = false;
+      this.selectedSuggestion = null;
       this.inputValue = '';
       this.onSubmit(value);
     },
     makeSelection(event) {
       this.inputValue = event.target.textContent;
       this.onFormSubmit();
     },
-  },
-
-  mounted() {
-    window.addEventListener('click', (event) => {
+    onWindowClick(event) {
       if (!getEventTarget(event.target, this.$el)) {
         this.showSuggestions = false;
+        this.selectedSuggestion = null;
+      }
+    },
+    onWindowKeyUp(event) {
+      if (this.$refs.input !== window.document.activeElement) {
+        return;
       }
-    });
+
+      switch (event.key) {
+        case 'Escape':
+          this.showSuggestions = false;
+          this.selectedSuggestion = null;
+          break;
+        case 'ArrowDown':
+          this.showSuggestions = true;
+
+          if (
+            this.selectedSuggestion === null ||
+            this.selectedSuggestion === this.suggestionsList.length - 1
+          ) {
+            this.selectedSuggestion = 0;
+          } else {
+            this.selectedSuggestion += 1;
+          }
+
+          break;
+        case 'ArrowUp':
```

Comment:
Please add a test case for usage of these keys
https://docs.cypress.io/api/commands/type

Reply:
Done https://github.com/applandinc/appmap-js/pull/492/commits/87e5beef37627ed4024836b8515e1fd3c85b8cda